cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/AuditMetadataStorage.java

```diff
@@ -63,24 +69,51 @@ public void setAuditPublisher(AuditPublisher auditPublisher) {
 
   @Override
   public void createIndex() throws IOException {
-    storage.createIndex();
+    try {
+      storage.createIndex();
+      emitMetrics("createIndex.count");
+    } catch (Exception e) {
+      emitMetrics("createIndex.error");
+      throw e;
+    }
   }
 
   @Override
   public void dropIndex() throws IOException {
-    storage.dropIndex();
+    try {
+      storage.dropIndex();
+      emitMetrics("dropIndex.count");
+    } catch (Exception e) {
+      emitMetrics("dropIndex.error");
+      throw e;
+    }
   }
 
   @Override
   public MetadataChange apply(MetadataMutation mutation) throws IOException {
-    MetadataChange change = storage.apply(mutation);
+    MetadataChange change;
+    try {
+      change = storage.apply(mutation);
+      emitMetrics("apply.count");
```

Comment:
there are four types of mutations: create, drop, update and remove. Do you think we could have separate metrics for them?